src/xdpinstaller/scripts/xdp-setup.ps1

```diff
@@ -0,0 +1,226 @@
+<#
+
+.SYNOPSIS
+This script installs or uninstalls various XDP components.
+
+.PARAMETER Install
+    Specifies an XDP component to install.
+
+.PARAMETER Uninstall
+    Attempts to uninstall all XDP components.
+
+#>
+
+param (
+    
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("", "xdp")]
+    [string]$Install = "",
+
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("", "xdp")]
+    [string]$Uninstall = ""
+)
+
+Set-StrictMode -Version 'Latest'
+$PSDefaultParameterValues['*:ErrorAction'] = 'Stop'
+
+# Global paths.
+$SystemFolder = [Environment]::SystemDirectory
+$XdpSys = "$SystemFolder\drivers\xdp.sys"
+$XdpInf = "$SystemFolder\drivers\xdp.inf"
+$XdpCat = "$SystemFolder\drivers\xdp.cat"
+$LogsDir = "$SystemFolder\Logs"
+
+# Set the temporary working directory and testore it on exit.
+Push-Location -Path $SystemFolder
+
+# Helper to reboot the machine.
+function Uninstall-Failure {
+    Write-Host "Capturing live kernel dump"
+
+    New-Item -ItemType Directory -Force -Path $LogsDir | Out-Null
+    Write-Verbose "$LiveKD -o $LogsDir\xdp.dmp -k $KD -ml -accepteula"
+    & $LiveKD -o $LogsDir\xdp.dmp -k $KD -ml -accepteula
+
+    Write-Host "##vso[task.setvariable variable=NeedsReboot]true"
+    Write-Error "Preparing to reboot machine!"
+}
+
+# Helper to start (with retry) a service.
+function Start-Service-With-Retry($Name) {
+    Write-Verbose "Start-Service $Name"
+    $StartSuccess = $false
+    for ($i=0; $i -lt 10; $i++) {
+        try {
+            Start-Sleep -Milliseconds 100
+            Start-Service $Name
+            $StartSuccess = $true
+            break
+        } catch { }
+    }
+    if ($StartSuccess -eq $false) {
+        Write-Error "Failed to start $Name"
+    }
+}
+
+# Helper to rename (with retry) a network adapter. On WS2022, renames sometimes
+# fail with ERROR_TRANSACTION_NOT_ACTIVE.
+function Rename-NetAdapter-With-Retry($IfDesc, $Name) {
```

Comment:
AFAICT this function is unused and can be safely removed.

Reply:
We can clean this up after the fact

Reply:
Removed.